lib/OpenQA/Schema/Result/JobModules.pm

```diff
@@ -90,10 +90,8 @@ sub results {
     return {} unless my $dir = $self->job->result_dir;
     my $name = $self->name;
 
-    return {} unless -r (my $file = path($dir, "details-$name.json"));
-
-    my $json_data = $file->slurp;
-    die qq{Malformed JSON file "$file": $@} unless my $json = eval { decode_json($json_data) };
+    my $file = path($dir, "details-$name.json");
+    die qq{Malformed/unreadable JSON file "$file": $@} unless my $json = eval { decode_json($file->slurp) };
```

Comment:
Supposedly slurp and decoding needed to be separated to handle those errors distinctively.

Reply:
Why? To have distinct error messages?